core/src/main/java/feign/SynchronousMethodHandler.java

```diff
@@ -105,14 +104,25 @@ Object executeAndDecode(RequestTemplate template) throws Throwable {
     }
     long elapsedTime = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - start);
 
-    boolean shouldClose = true;
     try {
       if (logLevel != Logger.Level.NONE) {
         response =
-            logger.logAndRebufferResponse(metadata.configKey(), logLevel, response, elapsedTime);
+                logger.logAndRebufferResponse(metadata.configKey(), logLevel, response, elapsedTime);
         // ensure the request is set. TODO: remove in Feign 10
         response.toBuilder().request(request).build();
       }
+      return decodeResponse(response);
```

Comment:
Moving this to another method just makes it harder to see what you are actually changing.